tools/stem_queries.cpp

```diff
@@ -0,0 +1,43 @@
+#include <fstream>
+#include <string>
+
+#include "pisa/query/queries.hpp"
+#include "pisa/query/term_processor.hpp"
+#include "pisa/tokenizer.hpp"
+#include <boost/algorithm/string/join.hpp>
+
+#include "CLI/CLI.hpp"
+#include "io.hpp"
+
+int main(int argc, char const* argv[])
+{
+    std::string input_filename;
+    std::string output_filename;
+    std::optional<std::string> stemmer;
+
+    CLI::App app{"A tool for stemming PISA queries."};
+    app.add_option("-i,--input", input_filename, "Query input file")->required();
+    app.add_option("-o,--output", output_filename, "Query output file")->required();
+    app.add_option("--stemmer", stemmer, "Stemmer")->required();
+
+    CLI11_PARSE(app, argc, argv);
+
+    std::ofstream output_file;
+    output_file.open(output_filename);
+
+    auto input_file = std::ifstream(input_filename);
+    pisa::io::for_each_line(input_file, [&](std::string const& query_string) {
```

Comment:
Can we move this lambda somewhere to the lib and write a test? I was thinking of maybe something like:
```cpp
try {
    LineStemmer line_stemmer(stemmer);
    pisa::io::for_each_line(input_file, [&](auto const& line) { line_stemmer.stem_line(line); });
} catch (whatever LineStemmer constructor throws) {
    spdlog::error(...);
    std::exit(1);
}
```
Or at least something like this. I'm not married to the names either.